src/main/scala/no/gdl/bookapi/service/search/SearchService.scala

```diff
@@ -0,0 +1,134 @@
+import no.gdl.bookapi.service.ConverterService
+import org.elasticsearch.ElasticsearchException
+import org.elasticsearch.index.IndexNotFoundException
+import org.elasticsearch.index.query.{BoolQueryBuilder, QueryBuilders}
+import org.json4s.DefaultFormats
+import org.json4s.native.Serialization.read
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+
+trait SearchService {
+  this: ElasticClient with ConverterService with IndexBuilderService with IndexService =>
+  val searchService: SearchService
+
+  class SearchService extends LazyLogging {
+    implicit val formats = DefaultFormats + LocalDateSerializer
+
+    def searchBook(query: Option[String], language: LanguageTag, page: Int, pageSize: Int): SearchResult =
+      executeSearch(QueryBuilders.boolQuery(), query, language, page, pageSize)
+
+    def executeSearch(queryBuilder: BoolQueryBuilder, query: Option[String], language: LanguageTag, page: Int, pageSize: Int): SearchResult = {
+
+      val (startAt, numResults) = getStartAtAndNumResults(Some(page), Some(pageSize))
+
+      val requestedResultWindow = page * numResults
+      if (requestedResultWindow > BookApiProperties.ElasticSearchIndexMaxResultWindow) {
+        logger.info(s"Max supported results are ${BookApiProperties.ElasticSearchIndexMaxResultWindow}, user requested ${requestedResultWindow}")
+        throw new ResultWindowTooLargeException(Error.WindowTooLargeError.description)
+      }
+
+      val indexAndTypes = new IndexAndTypes(BookApiProperties.searchIndex(language), Seq(BookApiProperties.SearchDocument))
+      val searchResponse = esClient.execute(
+        searchWithType(indexAndTypes).size(numResults).from(startAt)
+          .bool(BoolQueryDefinition()
+            .should(QueryStringQueryDefinition(query = query.getOrElse("*")).field("title").field("description")))
+          .sortBy(FieldSortDefinition("id"))
+      )
+
+      searchResponse match {
+        case Success(response) => SearchResult(response.result.totalHits, page, numResults, converterService.toApiLanguage(language), getHits(response.result.hits, language))
+        case Failure(failure) => errorHandler(Failure(failure))
+      }
+    }
+
+    def getHits(hits: SearchHits, language: LanguageTag): Seq[Book] = {
+      var resultList = Seq[Book]()
+      hits.total match {
+        case count: Long if count > 0 => {
+          val results = hits.hits
+          val iterator = results.iterator
+          while(iterator.hasNext) {
+            resultList = resultList :+ hitAsApiBook(iterator.next().sourceAsString, language)
+          }
+          resultList
+        }
+        case _ => Seq()
+      }
+    }
+
+    def hitAsApiBook(hit: String, language: LanguageTag): Book = {
+      read[Book](hit)
+    }
+
+    private def getStartAtAndNumResults(page: Option[Int], pageSize: Option[Int]): (Int, Int) = {
```

Comment:
Why are the arguments wrapped in `Option`? The function is only called with `Some`-values.